lowranceusr.cc

```diff
@@ -455,226 +1378,361 @@ lowranceusr_parse_routes()
     route_add_head(rte_head);
     rte_head->rte_num = i+1;
 
-    /* route name */
-    int text_len = lowranceusr_readstr(&buff[0], MAXUSRSTRINGSIZE, file_in);
-    if (text_len) {
-      buff[text_len] = '\0';
-      rte_head->rte_name = buff;
+    if (reading_version < 4) {
+      lowranceusr_parse_route();
+    } else {
+      lowranceusr4_parse_route();
     }
+  }
+}
+
 
-    /* num Legs */
-    short num_legs = gbfgetint16(file_in);
+/*
+ * Icons are automatically converted to waypoints unless
+ * option of ignoreicons is used
+ */
+static void
+lowranceusr_parse_icons()
+{
+  char buff[MAXUSRSTRINGSIZE + 1];
 
-    /* route reversed */
-    (void) gbfread(&buff[0], 1, 1, file_in);
+  short int num_icons = gbfgetint16(file_in);
 
-    /* waypoints */
-    for (int j = 0; j < num_legs; j++) {
+  if (global_opts.debug_level >= 1) {
+    printf(MYNAME " parse_icons: Num Event Marker Icons = %d\n", num_icons);
+  }
+
+  for (int i = 0; i < num_icons && !gbfeof(file_in); i++) {
+    double latitude    = lat_mm_to_deg(gbfgetint32(file_in));
+    double longitude   = lon_mm_to_deg(gbfgetint32(file_in));
+    int    icon_number = gbfgetint32(file_in);
+
+    if (opt_ignoreicons == nullptr) {          /* Option not specified if NULL */
       Waypoint* wpt_tmp = new Waypoint;
-      lowranceusr_parse_waypt(wpt_tmp);
-      route_add_wpt(rte_head, wpt_tmp);
+
+      /* position coord lat & long */
+      wpt_tmp->latitude = latitude;
+      wpt_tmp->longitude = longitude;
+      wpt_tmp->altitude = 0;
+      snprintf(buff, sizeof(buff), "Event Marker %d", i+1);
```

Comment:
...shortname = "Event marker" + QString::number (i+1);

In general, avoid s[n]sprintf when we can.